tools/rollback.sh

```diff
@@ -1,8 +1,29 @@
 #!/bin/bash
 
-touch "were it not for the laws of this land"
-# if gateway_old is less than 2 weeks old
-if [ -d "gateway_old" ]; then
+# This script performs a rollback of a failed upgrade. It expects to be run in
+# the ~/mozilla-iot directory where it can see gateway, gateway_old, and
+# gateway_failed
+
+# from https://stackoverflow.com/questions/552724/
+function recentEnough() {
+   local filename=$1
+   local changed=`stat -c %Y "$filename"`
+   local now=`date +%s`
+   local elapsed
+
+   let elapsed=now-changed
+   # if less than 60 * 60 * 24 * 14 seconds have passed
+   if [ $elapsed -lt 1209600 ]; then
+     return 1
+   else
+     return 0
+   fi
+}
+
+if [ -d "gateway_old" ] && [ ! -z `recentEnough "gateway_old"` ]; then
```

Comment:
nit: Personally, I prefer to use $(recentEnough ...) rather than backticks. I find the $() notation makes it much easier to match up when expressions get more complex.